packages/build-info/src/build-systems/nrwl.ts

```diff
@@ -1,27 +1,50 @@
+import type { PackageJson } from 'read-pkg'
+
+import { NPM_BUILD_SCRIPTS, NPM_DEV_SCRIPTS } from '../get-commands.js'
+import { WorkspaceInfo, WorkspacePackage } from '../workspaces/detect-workspace.js'
 import { findPackages, identifyPackageFn } from '../workspaces/get-workspace-packages.js'
 
-import { BaseBuildTool, Command } from './build-system.js'
+import { BaseBuildTool, type Command } from './build-system.js'
 
 export class Nx extends BaseBuildTool {
   id = 'nx'
   name = 'Nx'
   configFiles = ['nx.json']
-
-  build = {}
+  runFromRoot = true
+  /**
+   * if it's an nx integrated setup
+   * We need to differentiate as the 'dist' folder is located differently
+   * @see https://nx.dev/concepts/integrated-vs-package-based
+   */
+  isIntegrated = false
 
   /** Retrieves a list of possible commands for a package */
   async getCommands(packagePath: string): Promise<Command[]> {
-    const { name, targets } = await this.project.fs.readJSON(this.project.fs.join(packagePath, 'project.json'))
+    const projectPath = this.project.resolveFromPackage(packagePath, 'project.json')
+    const packageJSONPath = this.project.resolveFromPackage(packagePath, 'package.json')
+    let name: string
+    const targets: string[] = []
+    try {
+      const project = await this.project.fs.readJSON(projectPath, { fail: true })
+      this.isIntegrated = true
+      targets.push(...Object.keys(project?.targets || {}))
+      name = (project.name as string) || ''
+    } catch {
+      // if no project.json exists it's probably a package based nx workspace and not a integrated one
+      const json = await this.project.fs.readJSON<PackageJson>(packageJSONPath)
+      targets.push(...Object.keys(json?.scripts || {}))
+      name = json.name || ''
+    }
 
-    if (name && targets) {
-      return Object.keys(targets).map((target) => {
+    if (name.length && targets.length) {
```

Comment:
nit: I think being explicit makes things a bit clearer, but totally a nit. Feel free to take it or leave it!

```suggestion
    if (name.length !== 0 && targets.length !== 0) {
```

Reply:
Addressed in https://github.com/netlify/build/pull/4977/commits/ac28711e57721593b35986c1b39480354cba73af